pkg/set/set.go

```diff
@@ -188,6 +188,84 @@ func (s *Set) GetSet() *nftables.Set {
 	return s.set
 }
 
+// Gets a set's SetData with associated counters
+func (s *Set) getCountedSetData(c *nftables.Conn) ([]countedSetData, error) {
+	elements, err := c.GetSetElements(s.set)
+	if err != nil {
+		return nil, err
+	}
+
+	if s.set.KeyType == nftables.TypeIPAddr || s.set.KeyType == nftables.TypeIP6Addr {
+		return countedAddrSetData(elements)
+	} else if s.set.KeyType == nftables.TypeInetService {
+		return countedPortSetData(elements)
+	}
+
+	return nil, fmt.Errorf("unexpected set key type: %v", s.set.KeyType)
+}
+
+func countedPortSetData(elements []nftables.SetElement) ([]countedSetData, error) {
+	setDataList := []countedSetData{}
+
+	// set elements come in pairs, first the end of range, then start of range which contains counters
+	for i := 0; i < len(elements); i++ {
+		endElement := elements[i]
+		if !endElement.IntervalEnd {
```

Comment:
Maybe we can create a function that will return the start and end elements so we can use the same code in both `countedPortSetData` and `countedAddrSetData`?